docs/schema-definition.md

```diff
@@ -160,8 +160,64 @@ $schema = new Schema([
 You can automate this registry if you wish to reduce boilerplate or even
 introduce a Dependency Injection Container if your types have other dependencies.
 
-Alternatively, all methods of the registry could be static - then there is no need
-to pass it in the constructor - instead use **TypeRegistry::myAType()** in your type definitions.
+This approach has its drawbacks; while using a callable for the **fields** property does provide some protection, once the fields property is accessed each individual field under its governance is still evaluated whether it's needed for the current query or not, and this can result in many superfluous field and argument instantiations. For a more robust solution, you can skip the type loader altogether and instead craft a type registry populated with static helpers that return callables for each individual type:
```

Comment:
> skip the type loader altogether

I don't think that will work, unless the schema is constructed with `types`. Retrieving types by name is necessary in cases where the client defines a type name, e.g. variables, fragments.

Reply:
Seems to work just fine in my own code, but to be fair I didn't ensure I've covered all edge cases. For now I'll reword that...

Reply:
Is that for selecting on a union? That seems to work just fine, too, presumable because I use the same static helpers everywhere, eg:

```sdl
// a piece of a query from my code
fragment skuEdges on Skus {
    items {
      edges {
        ... on SkuRow {
          sku {
            ...smallSku
          }
        }
      }
    }
  }
```

```php
// my union resolver:
class SortedSkuType extends UnionType
{
    public function __construct()
    {
        parent::__construct([
            'types' => [
                Types::skuRow(),
                Types::sortSeparator()
            ],
            'resolveType' => static function ($value): callable {
                if (is_object($value)) {
                    return Types::skuRow();
                } else {
                    return Types::sortSeparator();
                }
            }
        ]);
    }
}
```

At any rate, I would not presume to push back on this too hard, as you know this stuff far better than me, and there's no harm in dropping the suggestion from the documentation, which I've already done.

Reply:
Have you tried exactly the query that I suggested? I think that looking for an unknown type would fail without a type loader. How do you construct your `Schema`?

Reply:
I'm not sure I understand what you're getting at with this. Yes, if I run your exact query without a type loader, I get an error:

```
{
  "errors": [
    {
      "message": "Unknown type \"SomeUnknownType\".",
      "locations": [
        {
          "line": 1,
          "column": 20
        }
      ],
      "code": 0,
      "extensions": {
        "file": "/var/www/html/vendor/webonyx/graphql-php/src/Validator/Rules/KnownTypeNames.php",
        "line": 87
      }
    }
  ]
}
```

I also get an error _with_ a type loader. 
```
{
  "error": {
    "message": "Unknown graphql type: SomeUnknownType"
  }
}
```

Other than one having nicer formatting than the other, it's not clear to me what we've proven, here. What is the use case for querying for types that don't exist?

Regardless, I removed the bit you didn't like 4 days ago. What is left to do?

Reply:
Just replying to myself, here--I was experimenting with removing the type loader again, and it's definitely not advised; removing it doesn't break anything, but it _does_ lead to invocation of `TypeInfo::extractTypes`, which tours the entire tree sniffing for types and is probably not what you want whether you're using lazy loading or not.

Reply:
Ah right, that makes sense. As of now, your new example does not function as a type loader - whereas the existing example at https://webonyx.github.io/graphql-php/schema-definition/#lazy-loading-of-types does. Can we get the best of both worlds? Also, I would like to avoid overwhelming readers with different implementation variants and rather just describe a single approach that is optimal for most use cases.

Reply:
Yeah, let me review how everything works again and then I'll rewrite the whole section. I would have done that in the first place but I didn't want to step on any toes. I'll work on it when I get a chance.